### ISSUE TITLE
Update dependencies for Angular 5

### ISSUE THREAD

**Issue by livthomas** [isaacplmann/ngx-contextmenu#72]
`ngx-contextmenu` seems to work with Angular 5.0 but there are some warnings in the output of `npm install`:
```
npm WARN ngx-contextmenu@2.0.0-beta.7 requires a peer of @angular/common@^4.0.0 but none was installed.
npm WARN ngx-contextmenu@2.0.0-beta.7 requires a peer of @angular/core@^4.0.0 but none was installed.
```

`@angular/cdk` dependecy version could also be updated to the latest `5.0.0-rc0`. Anyway, why is this dependency needed?

**Comment by isaacplmann** [comment]
I'm offloading the positioning logic to the `@angular/cdk`.  The `cdk` makes fallback positioning a lot easier and handles a lot of the browser inconsistencies for me.

This is fixed in PR #76 . I'll be publishing a new version later today.

**Comment by isaacplmann** [comment]
Fixed in version 3.0.0.  I had to drop support for angular 4.x because the latest version of `@angular/cdk` did.

**Comment by jfrankowski** [comment]
You should write that down in description. I have just lost a lot of time trying to get it work in angular4
